src/com/dojo/foundations/parking/TheGarage.java

```diff
@@ -89,81 +49,71 @@ public static void main(String[] args) throws Exception {
                     System.out.println("Write the motorcycle plate:");
                     String motorcyclePlate = scanner.readLine();
                     Matcher matcherMotorcycle = PATTERN_MOTORCYCLE_PLATE.matcher(motorcyclePlate);
-                    if (!matcherMotorcycle.find()) {
-                        System.out.println("The plate is invalid");
-                        return;
-                    }
+                    evaluatePlate(matcherMotorcycle);
                     System.out.println("Write the motorcycle cc:");
                     int motorcycleCc = Integer.parseInt(scanner.readLine());
 
                     Optional<Cell> availableCellMotorcycle = parkingLot.getParkingLot().stream()
-                            .filter(cell -> CellTypeEnum.MOTORCYCLE.equals(cell.getType()) && cell.isFree)
+                            .filter(cell -> CellTypeEnum.MOTORCYCLE.equals(cell.getType()) && cell.isFree())
```

Comment:
So the attribute 'isFree' is now private and we are using the intermediary getter instead of accessing it directly.  Pure encapsulation, great

Reply:
A private method here would be cool since several parts of the code run basically the same